azdev/operations/help/refdoc/cli_docs/helpgen.py

```diff
@@ -0,0 +1,53 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+import json
+
+from azdev.utilities import get_cli_repo_path
+from azdev.operations.help import DOC_SOURCE_MAP_PATH
+from azdev.operations.help.refdoc.common.directives import AbstractHelpGenDirective
+from azdev.operations.help.refdoc.common.directives import setup_common_directives
+
+from azure.cli.core._help import CliCommandHelpFile
+from azure.cli.core.file_util import create_invoker_and_load_cmds_and_args, get_all_help
+
+class HelpGenDirective(AbstractHelpGenDirective):
+    """ General CLI Sphinx Directive
+        The Core CLI has a doc source map to determine help text source for core cli commands. Extension help processed
+        here will have no doc source
+    """
+
+    def _get_help_files(self, az_cli):
+        create_invoker_and_load_cmds_and_args(az_cli)
+        return get_all_help(az_cli)
+
+    def _load_doc_source_map(self):
+        return {}
+        # todo: get_cli_repo_path() is failing for some reason.
```

Comment:
@tjprescott get_cli_repo_path() doesn't work for me.

I guess a work around would be for the users to specify doc_source_map's path as an argument to `azdev generate_docs`

Reply:
My b, I think i just needed to re-run azdev setup -c